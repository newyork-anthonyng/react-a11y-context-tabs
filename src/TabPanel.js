import React from "react";
import PropTypes from "prop-types";

const TabPanel = (
  { id, children, className, ...rest },
  { selectedId, forceRenderTabPanel, selectedTabPanelClassName }
) => {
  const isSelected = id === selectedId;

  let updatedClassName = className;
  if (isSelected) updatedClassName += ` ${selectedTabPanelClassName}`;

  return (
    <div
      id={`${id}--panel`}
      className={updatedClassName}
      aria-labelledby={`${id}--tab`}
      {...rest}
    >
      {(isSelected || forceRenderTabPanel) && children}
    </div>
  );
};

TabPanel.propTypes = {
  className: PropTypes.string,
  id: PropTypes.string.isRequired,
  children: PropTypes.node
};

TabPanel.defaultProps = {
  className: "react-tabs__tab-panel"
};

TabPanel.contextTypes = {
  selectedId: PropTypes.string,
  forceRenderTabPanel: PropTypes.bool,
  selectedTabPanelClassName: PropTypes.string
};

export default TabPanel;
