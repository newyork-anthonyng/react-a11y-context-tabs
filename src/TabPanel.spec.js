import React from "react";
import { mount } from "enzyme";
import TabPanel from "./TabPanel";
import toJSON from "enzyme-to-json";

const defaultProps = {
  id: "foo",
  className: "tabPanel"
};
const defaultContext = {
  selectedId: "bar",
  forceRenderTabPanel: false,
  selectedTabPanelClassName: "tabPanel--selected"
};

it("should render children when selected", () => {
  const wrapper = mount(
    <TabPanel {...defaultProps}>
      <h1>Hello</h1>
    </TabPanel>,
    {
      context: {
        ...defaultContext,
        selectedId: defaultProps.id
      }
    }
  );

  expect(toJSON(wrapper)).toMatchSnapshot();
});

it("should render children when forceRenderTabPanel is true", () => {
  const wrapper = mount(
    <TabPanel {...defaultProps}>
      <h1>Hello</h1>
    </TabPanel>,
    {
      context: {
        ...defaultContext,
        forceRenderTabPanel: true
      }
    }
  );

  expect(toJSON(wrapper)).toMatchSnapshot();
});

it("should not render children when not selected and forceRenderTabPanel is false", () => {
  const wrapper = mount(
    <TabPanel {...defaultProps}>
      <h1>Hello</h1>
    </TabPanel>,
    {
      context: defaultContext
    }
  );

  expect(toJSON(wrapper)).toMatchSnapshot();
});
