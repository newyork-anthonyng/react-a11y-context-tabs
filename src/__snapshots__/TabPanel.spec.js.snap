// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should not render children when not selected and forceRenderTabPanel is false 1`] = `
<TabPanel
  className="tabPanel"
  id="foo"
>
  <div
    aria-labelledby="foo--tab"
    className="tabPanel"
    id="foo--panel"
  />
</TabPanel>
`;

exports[`should render children when forceRenderTabPanel is true 1`] = `
<TabPanel
  className="tabPanel"
  id="foo"
>
  <div
    aria-labelledby="foo--tab"
    className="tabPanel"
    id="foo--panel"
  >
    <h1>
      Hello
    </h1>
  </div>
</TabPanel>
`;

exports[`should render children when selected 1`] = `
<TabPanel
  className="tabPanel"
  id="foo"
>
  <div
    aria-labelledby="foo--tab"
    className="tabPanel tabPanel--selected"
    id="foo--panel"
  >
    <h1>
      Hello
    </h1>
  </div>
</TabPanel>
`;
